src/vs/workbench/contrib/void/browser/react/src/sidebar-tsx/SidebarChat.tsx

```diff
@@ -1129,15 +1129,24 @@ const UserMessageComponent = ({ chatMessage, messageIdx, isCheckpointGhost, curr
 
 
 		<div
-			className="absolute -top-1 -right-1 translate-x-0 -translate-y-0 z-1"
+			className="absolute flex -top-1 -right-1 gap-1 translate-x-0 -translate-y-0 z-1"
 		// data-tooltip-id='void-tooltip'
 		// data-tooltip-content='Edit message'
 		// data-tooltip-place='left'
 		>
+			<CopyButton
+				className={`
+									bg-void-bg-1 border border-void-border-1 rounded-md
+									transition-opacity duration-200 ease-in-out
+									${isHovered && mode === 'display' ? 'opacity-100' : 'opacity-0 focus:opacity-0'}
+								`}
+				text={chatMessage.displayContent}
+				toolTipName='Copy'
+			/>
 			<EditSymbol
 				size={18}
 				className={`
-                    cursor-pointer
+                    cursor-pointer text-void-fg-3
```

Comment:
I added this to make the pencil icon less striking. Not sure if it fits your style. Can change if it doesn't